File: algorithms/easy/maximum_69_number/solution.py
class Solution:
    def maximum69Number (self, num: int) -> int:
        num_str, max_num = str(num), num
        for i in range(len(num_str)):
            if num_str[i] == '6':
                count = num + 3 * (10**(len(num_str)-i-1))
                if count > max_num:
                    max_num = count

        return max_num

solution = Solution()
print(solution.maximum69Number(9669) == 9969)
print(solution.maximum69Number(9996) == 9999)
print(solution.maximum69Number(9999) == 9999)

File: algorithms/easy/maximum_69_number/solution.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func maximum69Number(num int) int {
	str, maxNum := strconv.Itoa(num), num
	for i := 0; i < len(str); i++ {
		if string(str[i]) == "6" {
			count := num + 3*int(math.Pow(10, float64(len(str)-i-1)))
			if count > maxNum {
				maxNum = count
			}
		}
	}

	return maxNum
}

func main() {
	fmt.Println(maximum69Number(9669) == 9969)
	fmt.Println(maximum69Number(9996) == 9999)
	fmt.Println(maximum69Number(9999) == 9999)
}
